File: src/main/java/com/example/accountbook/repository/StatementCustomRepository.java
package com.example.accountbook.repository;

import java.time.LocalDateTime;
import java.util.List;

public interface StatementCustomRepository {

    List<Integer> sumAmountWeekly(Long mainCategoryId, LocalDateTime date);

    List<Integer> sumAmountMonthly(Long mainCategoryId, LocalDateTime data);
}

File: src/main/java/com/example/accountbook/repository/StatementCustomRepository.kt
package com.example.accountbook.repository

import com.querydsl.core.Tuple
import java.time.LocalDateTime

interface StatementCustomRepository {
    fun sumAmountWeekly(mainCategoryId: Long, date: LocalDateTime): List<Int>
    fun sumAmountMonthly(mainCategoryId: Long, date: LocalDateTime): List<Int>
    fun sumAmountWeeklyGroupByCategory(date: LocalDateTime?): List<Tuple>
    fun sumTotalAmountMonthly(categoryType: Int?, date: LocalDateTime): List<Int>
    fun sumTotalDiscountMonthly(date: LocalDateTime): List<Int>
    fun sumTotalSavingMonthly(date: LocalDateTime): List<Int>

    fun sumStatementSummary(
        type: Int?,
        dateGte: LocalDateTime?,
        dateLte: LocalDateTime?,
        categoryId: Long?,
        mainCategoryId: Long?
    ): List<Tuple>
}
